### ISSUE TITLE
Payment Data Class Typecast

### ISSUE THREAD

**Issue by Himanshu0047** [PaystackHQ/checkout-android#13]
I am facing this issue in the checkoutResultListener in the onError method

**Issue:-**  com.squareup.moshi.JsonDataException: Expected an int but was 2396035915 at path $.data.id

**Response:-**
{"status":true,"message":"Transaction initialized","data":{"status":"success","message":"successful","access_code":"ky55egezwiperel","id":2396035915,"amount":100,"plan":"","email":"sharmahimanshu.7699@gmail.com","merchant_name":"Viago Inc","merchant_logo":"https://public-files-paystack-prod.s3.eu-west-1.amazonaws.com/integration-logos/paystack.jpg","currency":"GHS","testmode":true,"original_amount":100,"added_fees":0,"merchant_id":893465,"merchant_key":"pk_test_8515ab2a469baba33eae8adb797c738847aa4940","domain":"test","transaction_status":"abandoned","log":null,"custom_filters":0,"customer":{"id":"106354627","first_name":"","last_name":"","email":"sharmahimanshu.7699@gmail.com","phone":"","metadata":null,"created_at":"2022-12-21 12:02:29","updated_at":"2022-12-21 12:02:29"},"plan_details":null,"channels":["card","mobile_money"],"channel_options":{"mobile_money":[{"key":"MTN","value":"MTN","isNew":false,"phoneNumberRegex":"^\\+233(5(5|4|9)|2(4|5))\\d{7}$","phoneNumberPlaceholder":"050 000 0000"},{"key":"ATL","value":"Airtel/Tigo","isNew":false,"phoneNumberRegex":"^\\+233(5(6|7)|2(6|7))\\d{7}$","phoneNumberPlaceholder":"050 000 0000"},{"key":"VOD","value":"Vodafone","isNew":false,"phoneNumberRegex":"^\\+233(50|20)\\d{7}$","phoneNumberPlaceholder":"050 000 0000"}]},"merchant_channel_settings":{},"elevy_percentage":0.015,"is_remember_me_enabled":null,"skip_pin_check_for_card_types":["visa"]}}

**Comment by imKarthikMuru** [comment]
Hi Have you find the solution for this, am facing this issue too

**Comment by Himanshu0047** [comment]
> Hi Have you find the solution for this, am facing this issue too

Yes I downloaded the dependency as a module and made the necessary changes and it is working. For this issue I changed the data type of data.id from Int to Long

**Comment by imKarthikMuru** [comment]
Thanks for the help
